core/src/banking_stage.rs

```diff
@@ -1204,68 +1168,78 @@ impl BankingStage {
 
-        let mut transactions_execute_and_record_status: Vec<bool> = execution_results
-            .iter()
-            .map(|execution_result| execution_result.was_executed())
-            .collect();
+        let transactions_attempted_execution_count = execution_results.len();
+        let (executed_transactions, execution_results_to_transactions_time): (Vec<_>, Measure) =
+            Measure::this(
+                |_| {
+                    execution_results
+                        .iter()
+                        .zip(batch.sanitized_transactions())
+                        .filter_map(|(execution_result, tx)| {
+                            if execution_result.was_executed() {
+                                Some(tx.to_versioned_transaction())
+                            } else {
+                                None
+                            }
+                        })
+                        .collect()
+                },
+                (),
+                " execution_results_to_transactions",
+            );
 
         let (freeze_lock, freeze_lock_time) =
             Measure::this(|_| bank.freeze_lock(), (), "freeze_lock");
         execute_and_commit_timings.freeze_lock_us = freeze_lock_time.as_us();
 
         let (record_transactions_summary, record_time) = Measure::this(
-            |_| {
-                Self::record_transactions(
-                    bank.slot(),
-                    batch.sanitized_transactions(),
-                    &execution_results,
-                    poh,
-                )
-            },
+            |_| Self::record_transactions(bank.slot(), executed_transactions, poh),
             (),
             "record_transactions",
         );
         execute_and_commit_timings.record_us = record_time.as_us();
 
         let RecordTransactionsSummary {
-            result: commit_transactions_result,
-            retryable_indexes: retryable_record_transaction_indexes,
+            result: record_transactions_result,
             record_transactions_timings,
         } = record_transactions_summary;
-        execute_and_commit_timings.record_transactions_timings = record_transactions_timings;
+        execute_and_commit_timings.record_transactions_timings = RecordTransactionsTimings {
+            execution_results_to_transactions_us: execution_results_to_transactions_time.as_us(),
+            ..record_transactions_timings
+        };
 
-        // mark transactions that were executed but not recorded
-        retryable_record_transaction_indexes.iter().for_each(|i| {
-            transactions_execute_and_record_status[*i] = false;
-        });
+        if let Err(recorder_err) = record_transactions_result {
+            let retryable_record_transaction_indexes: Vec<usize> = execution_results
+                .iter()
+                .enumerate()
+                .filter_map(|(index, execution_result)| {
+                    execution_result.was_executed().then(|| index)
```

Comment:
could probably directly push into `retryable_transaction_indexes` and then for the 
```
inc_new_counter_info!(
 "banking_stage-record_transactions_retryable_record_txs",
retryable_record_transaction_indexes.len()
);
```

counter below, just increment by the `executed_transactions_count`

Reply:
Yeah we could but I wanted to preserve the behavior of the counter where it only counts transactions that need to be retried due to record failure

Reply:
`executed_transactions_count` is equivalent to this right: https://github.com/solana-labs/solana/blob/67d3210ebd9055e78ef217c7c6e61559914731f1/core/src/banking_stage.rs#L1222, i.e. the number of transactions that were executed

Reply:
Ah yeah, you're right!